ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1458,14 +1459,61 @@ public void updateParentRequestQcState(String requestId, Integer qcStateId)
         }
     }
 
+    // Get all datasets that have a requestid column
+    private List<TableInfo> getRequestDatasets(Container c, User u)
```

Comment:
Hi @labkey-martyp: this doesnt surprise me that studydata is very slow. Two comments:

- I dont know how to do it better, but I believe at minimum ONPRC (and maybe other centers) have taskId, requestId, etc., added to all/most datasets. EHRManager.ensureDatasetPropertyDescriptors() does this. Even if all datasets have the property for some centers, separating this query probably makes sense.
- Regardless of the answer above, would it make sense to loop datasets and execute the SQL and update once per table, rather that making any kind of union SQL? Each query should be able to use the requestId index.

Reply:
I'm curious on whether there's a perf difference between this FB and Ben's. In general it should be somewhat faster to do one query instead of dozens, but it all comes down to what query plan the DB chooses.

Reply:
I tested both out and they were comparable in perf. Both a big improvement over using StudyData. A single union query was a little faster on the db side so I stuck with that.